keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -171,44 +173,47 @@ class ResourceActuator(
-            log.info("Resource {} delta: {}", id, diff.toDebug())
-            publisher.publishEvent(ResourceDeltaDetected(resource, diff.toDeltaJson(), clock))
+              plugin.create(resource, diff)
+                .also { tasks ->
+                  publisher.publishEvent(ResourceActuationLaunched(resource, plugin.name, tasks, clock))
+                  diffFingerprintRepository.markActionTaken(id)
+                }
+            }
+            diff.hasChanges() -> {
+              log.warn("Resource {} is invalid", id)
+              log.info("Resource {} delta: {}", id, diff.toDebug())
+              publisher.publishEvent(ResourceDeltaDetected(resource, diff.toDeltaJson(), clock))
 
-            plugin.update(resource, diff)
-              .also { tasks ->
-                publisher.publishEvent(ResourceActuationLaunched(resource, plugin.name, tasks, clock))
-                diffFingerprintRepository.markActionTaken(id)
-              }
-          }
-          else -> {
-            log.info("Resource {} is valid", id)
-            val lastEvent = resourceRepository.lastEvent(id)
-            when (lastEvent) {
-              is ResourceActuationLaunched -> log.debug("waiting for actuating task to be completed") // do nothing and wait
-              is ResourceDeltaDetected, is ResourceTaskSucceeded, is ResourceTaskFailed -> {
-                // if a delta was detected and a task wasn't launched, the delta is resolved
-                // if a task was launched and it completed, either successfully or not, the delta is resolved
-                publisher.publishEvent(ResourceDeltaResolved(resource, clock))
+              plugin.update(resource, diff)
+                .also { tasks ->
+                  publisher.publishEvent(ResourceActuationLaunched(resource, plugin.name, tasks, clock))
+                  diffFingerprintRepository.markActionTaken(id)
+                }
+            }
+            else -> {
+              log.info("Resource {} is valid", id)
+              val lastEvent = resourceRepository.lastEvent(id)
+              when (lastEvent) {
+                is ResourceActuationLaunched -> log.debug("waiting for actuating task to be completed") // do nothing and wait
+                is ResourceDeltaDetected, is ResourceTaskSucceeded, is ResourceTaskFailed -> {
+                  // if a delta was detected and a task wasn't launched, the delta is resolved
+                  // if a task was launched and it completed, either successfully or not, the delta is resolved
+                  publisher.publishEvent(ResourceDeltaResolved(resource, clock))
+                }
+                else -> publisher.publishEvent(ResourceValid(resource, clock))
               }
-              else -> publisher.publishEvent(ResourceValid(resource, clock))
             }
           }
+        } else {
+          log.info("Resource {} skipped because it can't be fixed: ${actionResponse.message}", id)
```

Comment:
Minor point but this might be a `warn` kinda thing, maybe?

Reply:
Good point, will update